src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/CachePoolClearerPass.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class CachePoolClearerPass implements CompilerPassInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        foreach ($container->findTaggedServiceIds('cache.pool') as $id => $tags) {
+            foreach ($tags as $attr) {
+            }
+            if (isset($attr['clearer'])) {
+                $clearer = $container->getDefinition($attr['clearer']);
+                $clearer->addMethodCall('addPool', array(new Reference($id)));
```

Comment:
The compiler pass is already implementation-aware, because `addPool` is not part of any interface as I can see. It's more like `Psr6CacheClearerPass` to be honest.

And I think it's OK for compiler pass to be locked on specific implementation, DIC is the place for this knowledge.